Update and simplify plugin structure

pfefferle opened the pull request:
## Summary

This PR modernizes the plugin development infrastructure:

- **Remove Grunt** - Replace with npm scripts for CSS compilation
- **Remove languages folder** - Translations handled by WordPress.org
- **Update composer.json** - Modern dependencies (PHP 7.2+, PHPUnit 9.x, WPCS 3.x)
- **Update phpcs.xml** - Clean ruleset with PHPCompatibilityWP, WordPress, VariableAnalysis
- **Add wp-env support** - Local WordPress development environment
- **Update GitHub workflows** - Modern action versions, proper triggers

## Changes

| Area | Change |
|------|--------|
| Build | Grunt → npm scripts (`build-css`, `build-icons`) |
| Dependencies | PHP 5.6 → 7.2+, updated dev dependencies |
| Linting | WPCS 3.x, PHPCompatibility 2.x, VariableAnalysis |
| Testing | PHPUnit 9.x, wp-env support |
| CI | Updated actions to v4, added pull_request triggers |

## New npm scripts

```bash
npm run build          # Build icons + CSS
npm run build-css      # Compile SCSS only
npm run wp-env:start   # Start local WordPress environment
npm run test:php       # Run PHPUnit via wp-env
npm run lint:php       # Run PHPCS via wp-env
```

## New composer scripts

```bash
composer lint          # Run PHPCS
composer lint:fix      # Run PHPCBF
composer test          # Run PHPUnit
composer test:wp-env   # Run PHPUnit via wp-env
```

## Test plan

- [ ] Verify CSS compilation works (`npm run build-css`)
- [ ] Verify icon generation works (`npm run build-icons`)
- [ ] Verify wp-env starts correctly
- [ ] Verify PHPUnit tests pass after merge (workflows will trigger on future PRs)
- [ ] Verify PHPCS passes after merge

pfefferle committed:
Update and simplify plugin structure

dshanske reviewed (changes requested):
The removal of SASS means the rebuild process for the icons is not scripted. The build icons script doesn't generate the final css file.

Probably could come up with a non Grunt solution though.

pfefferle commented:
Oooos! I will add an npm task instead! Good catch!

dshanske commented:
That build process is something I have thought of redoing. SinpleIcons has gotten a lot larger since I picked it and maybe I should be doing a subset.

Just curation is a pain.

dshanske commented:
I was thinking about this. I could add sass as a script and dependency to the package.json file and run it that way. Will test.

pfefferle committed:
Add npm script for CSS compilation

Replace Grunt sass task with npm script using Dart Sass:
- Add sass as devDependency
- Add build-css script that compiles SCSS to CSS with compressed output
- Add build script that runs both build-icons and build-css
- Regenerate CSS files using new build system

The build-css script compiles:
- sass/main.scss → static/css/indieweb.css
- sass/min.scss → static/css/indieweb-minimal.css
- sass/main-bw.scss → static/css/indieweb-bw.css
- sass/main-admin.scss → static/css/indieweb-admin.css

pfefferle committed:
Remove languages folder

pfefferle committed:
Add phpcs, phpunit and wp-env support

- Update composer.json with cleaner structure and modern dependencies
  - PHP 7.2+ requirement
  - PHPUnit 9.x, WPCS 3.x, PHPCompatibility 2.x
  - Simplified scripts for lint, lint:fix, test, test:wp-env
- Update phpcs.xml with modern ruleset structure
  - PHPCompatibilityWP, WordPress, VariableAnalysis rules
  - Minimum WP version 6.2
- Add .wp-env.json for local development environment
- Update package.json with wp-env scripts
  - wp-env:start, wp-env:stop
  - test:php, lint:php, lint:php:fix via wp-env
- Update phpunit.xml.dist with coverage configuration
- Update tests/bootstrap.php for better polyfills path detection

pfefferle committed:
Update GitHub workflows

phpunit.yml:
- Update actions to v4
- Add path filtering for efficiency
- Run on pull_request events
- Test PHP 7.2, 7.4, 8.0, 8.2, 8.3
- Test WP 6.2, latest, and trunk
- Use bin/install-wp-tests.sh directly

phpcs.yml:
- Update actions to v4
- Add path filtering
- Run on pull_request events
- Use PHP 8.2
- Output checkstyle format with cs2pr for PR annotations

deploy.yml / update-assets.yml:
- Update checkout action to v4

pfefferle committed:
Merge trunk and resolve conflicts

pfefferle committed:
Bump minimum PHP version to 7.4 (required by PHPUnit 9 and WP 6.2)

pfefferle committed:
Improve code formatting and readability

This commit adjusts code formatting across several files for better readability and consistency, including improved indentation, spacing, and alignment of function arguments. No functional changes were made.

pfefferle committed:
Fix PHPCS coding standards violations

- Add file doc comments to all PHP files
- Add function/method doc comments with proper formatting
- Fix inline comments to end with full stops
- Replace _e() with wp_kses() for HTML content in getting-started.php
- Add proper escaping to h-card.php template
- Convert /* style comments to /** doc comments in plugin-installer.php
- Add phpcs:ignore for necessary exceptions (template variables, API contracts)
- Fix comment capitalization and formatting throughout

pfefferle committed:
Fix PHPCS warnings across plugin files

- Remove unused function parameters and variables
- Reduce filter argument count where full args aren't needed
- Rename reserved keyword parameter ($string -> $domain_string)
- Add phpcs:ignore for WordPress API required parameters
- Add phpcs:disable/enable for template include variables
- Simplify nested conditionals in plugin installer
- Auto-fix alignment issues

pfefferle committed:
Install subversion in PHPUnit workflow for test setup

pfefferle committed:
Fix security escaping in plugin installer

- Sanitize POST variables with sanitize_text_field() and sanitize_key()
- Add wp_unslash() and isset() checks for POST data
- Escape plugin author output with wp_kses()
- Replace die() with wp_die()

dshanske reviewed (approved):
This looks good to me.
